utils/map.go

```diff
@@ -0,0 +1,45 @@
+package utils
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+func GetKeyValuesToUpdate(tags map[string]string, upstreamTags map[string]string) map[string]*string {
+	if len(tags) == 0 {
+		return nil
+	}
+
+	if len(upstreamTags) == 0 {
+		return aws.StringMap(tags)
+	}
+
+	updateTags := make(map[string]*string)
+	for key, val := range tags {
+		if upstreamTags[key] != val {
+			updateTags[key] = aws.String(val)
+		}
+	}
+
+	if len(updateTags) == 0 {
+		return nil
+	}
+	return updateTags
+}
+
+func GetKeysToDelete(tags map[string]string, upstreamTags map[string]string) []*string {
+	if len(upstreamTags) == 0 {
+		return nil
+	}
+
+	var updateUntags []*string
+	for key, val := range upstreamTags {
+		if len(tags) == 0 || tags[key] != val {
```

Comment:
if len(tags) is 0, do you think we can check this before the for loop and return nil?

Reply:
No because if len(tags) is 0 then we want to delete all of the keys.

Reply:
oh I misread this, makes sense